control/controller.go

```diff
@@ -65,22 +78,48 @@ type Config struct {
 	ExecutorDependencies execute.Dependencies
 }
 
+func (c *Config) Validate() error {
+	if c.ConcurrencyQuota <= 0 {
```

Comment:
This was originally my thought too, but @nathanielc asked that we require client code the be explicit about the resources limits so that behavior is clear.

Reply:
I think this is an anti-pattern in Go since a lot of Go generally assumes that it is ok for a struct not to have values assigned. There's no way to make this typesafe rather than to either add unnecessary errors or to panic.

At the same time, this can be fixed later so I don't really care that much. We'll add a way to make this easier to configure later. It might be good to use the `Option interface` pattern and to make `control.New` use whatever we decide the defaults should be at some point in the future. This is fine for now.